base/array.jl

```diff
@@ -653,6 +653,11 @@ function lexcmp(a::Array{UInt8,1}, b::Array{UInt8,1})
     return c < 0 ? -1 : c > 0 ? +1 : cmp(length(a),length(b))
 end
 
+# use memcmp for == on integer types
+=={T<:Union{Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64}}(a::Array{T,1}, b::Array{T,1}) =
+    (len = length(a)) == length(b) &&
+    ccall(:memcmp, Int32, (Ptr{T}, Ptr{T}, UInt), a, b, len * sizeof(T)) == 0
```

Comment:
I wonder if it's worth providing an `unsafe_memcmp` function, with the same signature as `unsafe_copy!`, to abstract away some of these details. I think the compiler would be good enough to inline `unsafe_memcmp`? If not, I'm a bit reluctant to do that, since this method will be used by comparing small strings, and that is a substantial performance issue in a lot of biotechnology applications.